src/rpc/mining.cpp

```diff
@@ -948,6 +964,107 @@ static UniValue estimaterawfee(const JSONRPCRequest& request)
     return result;
 }
 
+//
+// signet
+//
+
+UniValue getnewblockhex(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"getnewblockhex",
+        "\nGets hex representation of a proposed, unmined new signet block.\n",
+        {
+            {"coinbase_destination", RPCArg::Type::STR, RPCArg::Optional::NO, "Pay-out destination, as an address or a custom coinbase script"},
+        },
+        RPCResult{
+            "blockhex      (hex) The block hex\n"
+        },
+        RPCExamples{
+            HelpExampleCli("getnewblockhex", "bc1qkallealmkdwwyuc2tmf5c6hzmlaujq6jl38hpe")
+        }
+    }.Check(request);
+
+    if (!Params().GetConsensus().signet_blocks) {
+        throw std::runtime_error("getnewblockhex can only be used with signet networks");
+    }
+
+    CScript coinbase_script;
+    auto str = request.params[0].get_str();
+    if (IsValidDestinationString(str)) {
+        auto dest = DecodeDestination(str);
+        coinbase_script = GetScriptForDestination(dest);
+    } else {
+        util::insert(coinbase_script, ParseHexV(str, "coinbase_script"));
+    }
+
+    auto block = BlockAssembler(Params()).CreateNewBlock(coinbase_script)->block;
+    unsigned int extra_nonce = 0;
```

Comment:
The scope for `extra_nonce` could be reduced.

Reply:
It doesn't feel cleaner to put it inside the lock brackets as those are meant to do the lock, not limit scope. Unrelated, but I am considering switching to using `GetTimeMillis()` instead of `0`, to make multiple calls to grindblock actually get different results. Thoughts welcome, since your eyes were here recently. :)